libcontainer/cgroups/fscommon/rdma_test.go

```diff
@@ -0,0 +1,59 @@
+package fscommon
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/opencontainers/runc/libcontainer/configs"
+)
+
+/* Roadmap for future */
+// (Low-priority)  TODO: Check if it is possible to virtually mimic an actual RDMA device.
+// TODO: Think of more edge-cases to add.
+
+// TestRdmaSet performs an E2E test of RdmaSet(), parseRdmaKV() using dummy device and a dummy cgroup file-system.
+// Note: Following test does not gurantees that your host supports RDMA since this mocks underlying infrastructure.
+func TestRdmaSet(t *testing.T) {
+	tempDir := t.TempDir()
+	root := tempDir
+	testCgroupPath := filepath.Join(root, "rdma")
+
+	// Ensure the full mock cgroup path exists.
+	err := os.MkdirAll(testCgroupPath, 0o755)
```

Comment:
as `testCgroupPath` is a direct subdirectory of `root`, MkdirAll is not needed, it's plain old `Mkdir`.